model/states.go

```diff
@@ -14,70 +14,36 @@
 
 package model
 
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
 // StateType ...
 type StateType string
 
 const (
 	// StateTypeDelay ...
-	StateTypeDelay = "delay"
+	StateTypeDelay StateType = "delay"
 	// StateTypeEvent ...
-	StateTypeEvent = "event"
+	StateTypeEvent StateType = "event"
 	// StateTypeOperation ...
-	StateTypeOperation = "operation"
+	StateTypeOperation StateType = "operation"
 	// StateTypeParallel ...
-	StateTypeParallel = "parallel"
+	StateTypeParallel StateType = "parallel"
 	// StateTypeSwitch ...
-	StateTypeSwitch = "switch"
+	StateTypeSwitch StateType = "switch"
 	// StateTypeForEach ...
-	StateTypeForEach = "foreach"
+	StateTypeForEach StateType = "foreach"
 	// StateTypeInject ...
-	StateTypeInject = "inject"
+	StateTypeInject StateType = "inject"
 	// StateTypeCallback ...
-	StateTypeCallback = "callback"
+	StateTypeCallback StateType = "callback"
 	// StateTypeSleep ...
-	StateTypeSleep = "sleep"
+	StateTypeSleep StateType = "sleep"
 )
 
-func getActionsModelMapping(stateType string) (State, bool) {
-	switch stateType {
-	case StateTypeDelay:
-		return &DelayState{}, true
-	case StateTypeEvent:
-		return &EventState{}, true
-	case StateTypeOperation:
-		return &OperationState{}, true
-	case StateTypeParallel:
-		return &ParallelState{}, true
-	case StateTypeSwitch:
-		return &SwitchState{}, true
-	case StateTypeInject:
-		return &InjectState{}, true
-	case StateTypeForEach:
-		return &ForEachState{}, true
-	case StateTypeCallback:
-		return &CallbackState{}, true
-	case StateTypeSleep:
-		return &SleepState{}, true
-	}
-	return nil, false
-}
-
-// State definition for a Workflow state
-type State interface {
```

Comment:
Don't we need this interface anymore?

Reply:
no, which will fix https://github.com/serverlessworkflow/sdk-go/issues/144